src/Mvc/Mvc.Api.Analyzers/src/ActualApiResponseMetadataFactory.cs

```diff
@@ -58,23 +58,68 @@ void AnalyzeResponseExpression(IReturnOperation returnOperation)
         return allReturnStatementsReadable;
     }
 
-    internal static ActualApiResponseMetadata? InspectReturnOperation(
+    internal static List<ActualApiResponseMetadata> InspectReturnOperation(
         in ApiControllerSymbolCache symbolCache,
         IReturnOperation returnOperation)
     {
+        var apiResponses = new List<ActualApiResponseMetadata>();
+
         var returnedValue = returnOperation.ReturnedValue;
         if (returnedValue is null || returnedValue is IInvalidOperation)
         {
-            return null;
+            return apiResponses;
         }
 
-        // Covers conversion in the `IActionResult GetResult => NotFound()` case.
-        // Multiple conversions can happen for ActionResult<T>, hence a while loop.
-        while (returnedValue is IConversionOperation conversion)
+        // Add the original returned value to the stack as starting point.
+        // If there are conditional expressions, multiple operations are added to the stack.
+        // Due to this, it is possible to return multiple response metadata elements.
+        var pendingOperations = new Stack<IOperation>();
+        pendingOperations.Push(returnedValue);
+
+        while (pendingOperations.Any())
```

Comment:
Would this block be better authored using the visitor pattern? Roslyn's APIs already provide one for symbols, and it make it easier to expand the range of supported scenarios without adding too many if-else complexities.

Reply:
Sorry for the late reply, thanks for the good idea, will look into it when there is time 👍